pkg/ccl/changefeedccl/changefeed_processors.go

```diff
@@ -1272,6 +1272,7 @@ func (cf *changeFrontier) manageProtectedTimestamps(
 		return nil
 	}
 	cf.lastProtectedTimestampUpdate = timeutil.Now()
+	cf.metrics.ProtectedTimeStamp = cf.lastProtectedTimestampUpdate
```

Comment:
This is not safe: metrics are global and there could be multiple changefeeds running
on the server.

Reply:
If there is a shared metric between multiple change feeds and each feed can be attached to a unique ID:

```go
ProtectedTimeStamp struct {
    syncutil.Mutex
    TimeStamps map[int]time.Time
}
```

Use the oldest change feed time stamp as the data for the gauge.